### PR TITLE
Pie chart first pass

### PR DESCRIPTION
**Pull request by kevinylee1991**
First pass at the pie chart -- still need to make some tweaks like adding it to the appropriate cards and support for more props. Need to talk to madhura about the appropriate statistics for each card.

Screen Shot 2019-12-19 at 6 31 28 PM

### CODE REVIEW

File: src/app/core/components/graphs/PieGraph.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { PieChart, Pie, Cell } from 'recharts'
+import { useTheme } from '@material-ui/core/styles'
+
+const PieGraph = ({ data, sideLength, arcWidth, percent, primary, ...rest }) => {
```
Comment (jruddell) [inline src/app/core/components/graphs/PieGraph.js]: Can we make this a `.tsx` file?

**Comment by kevinylee1991** [comment]
Screen Shot 2020-01-03 at 1 05 09 PM

File: src/app/core/components/graphs/PieGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+// import PropTypes from 'prop-types'
+import { PieChart, Pie, Cell } from 'recharts'
+import { useTheme } from '@material-ui/core/styles'
+
+interface Props {
+  data: any
```
Comment (jruddell) [inline src/app/core/components/graphs/PieGraph.tsx]: can we add typings here for what the type of data is?

File: src/app/core/components/widgets/PieUsageWidget.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    padding: '6px 3px',
+    textAlign: 'right',
+  },
+  legendName: {
+    padding: '6px',
+  }
+}))
+
+// Primary is a string hat refers to the main stat to show percentage in middle of graph (ex. healthy)
+// data is an array of objects with format
```
Comment (jruddell) [inline src/app/core/components/widgets/PieUsageWidget.js]: Make this a typescript file, include an interface for what the legend wants. This way you can omit comments like this and help enforce its usage.

File: src/app/core/components/widgets/PieUsageWidget.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        percent={percent}
+        primary={primary}
+        {...rest}
+      />
+      <PieLegend data={data} marginLeft="20px" />
+    </div>
+  )
+}
+
+PieUsageWidget.propTypes = {
```
Comment (jruddell) [inline src/app/core/components/widgets/PieUsageWidget.js]: dont need propTypes anymore with typescript

File: src/app/core/components/widgets/PieUsageWidget.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  primary: PropTypes.string,
+  data: PropTypes.arrayOf(PropTypes.shape({
+    value: PropTypes.number,
+    name: PropTypes.string,
+    color: PropTypes.string,
+  })),
+}
+
+PieUsageWidget.defaultProps = {
```
Comment (jruddell) [inline src/app/core/components/widgets/PieUsageWidget.js]: dont use defaultProps please. You can just define the defaults when destructuring (less mental model to apply if we are trying to figure out what the defaults are).  `const PieUsageWidget = ({ primary, data, sideLength = 110, arcWidth = 12, ...rest }) => {`

File: src/app/plugins/kubernetes/components/dashboard/StatusCard.tsx [hunk 6]
```diff
@@ -90,17 +99,20 @@ interface StatusCardProps {
   icon: string | PropertyFunction<JSX.Element>
   quantity: number
   dataLoader: [() => any, {}] // todo figure out typings here.
-  quantityFn(data: any[]): { quantity: number, working: number, pending: number }
+  quantityFn(data: any[]): { quantity: number, pieData: any, piePrimary: string }
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/dashboard/StatusCard.tsx]: shouldn't use `any` type unless you absolutely need to. You know what `pieData` is but we dont, this is a great use case for an `interface` definition.

**Pull request merged**